napari/_vispy/utils_gl.py

```diff
@@ -58,10 +58,10 @@ def get_max_texture_sizes() -> Tuple[int, int]:
         max_size_2d = None
 
     # vispy doesn't expose GL_MAX_3D_TEXTURE_SIZE so hard coding for now.
-    # MAX_TEXTURE_SIZE_3D = gl.glGetParameter(gl.GL_MAX_3D_TEXTURE_SIZE)
-    # if MAX_TEXTURE_SIZE_3D == ():
-    #    MAX_TEXTURE_SIZE_3D = None
-    max_size_3d = 2048
+    GL_MAX_3D_TEXTURE_SIZE = 32883
+    max_size_3d = gl.glGetParameter(GL_MAX_3D_TEXTURE_SIZE)
```

Comment:
@JoOkuma what is actually happening here with the `GL_MAX_3D_TEXTURE_SIZE = 32883`? Does that become a default value?

Also if `max_size_3d = None` does the downstream code still work? I guess we do no downsampling then?

Also is the comment `# vispy doesn't expose GL_MAX_3D_TEXTURE_SIZE so hard coding for now.` now out of date?

Can you maybe add some comments to explain how this is working now

Reply:
I updated the comments, the magic number is just the value to access this parameter, the actual value is according to the driver.
Reference to the constants:
http://pyopengl.sourceforge.net/documentation/pydoc/OpenGL.GL.html

Reply:
ok i get it now, i might adjust the comment to be

```
# vispy/gloo doesn't provide the GL_MAX_3D_TEXTURE_SIZE location, but it can be found in this
# list of constants http://pyopengl.sourceforge.net/documentation/pydoc/OpenGL.GL.html
```

Also can you put this call inside our `_opengl_context` like the 2D case. This can be needed if someone wants to start up in 3D mode